### PR TITLE
Pin crate versions

### LINKED ISSUE #37: List of supported crates should also include supported version(s)
When authoring an exercise which requires a dependency, we could reduce guesswork by including the list of supported versions in the `supported_crates` file. This means that an exercise author can simply specify the appropriate version directly from the file, instead of naming the most recent version and hoping.

### PR DESCRIPTION
**Pull request by senekor**
closes #37 

This is picking up where #40 left off, essentially replacing the script & github action to update the dependencies with a dependabot config.

I've never made a dependabot config, so this is just what I think should work based on reading [these docs](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file).

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+updates:
+  - package-ecosystem: 'cargo'
+    directory: '/local-registry'
+    schedule:
+      interval: 'weekly'
+    groups:
+      - all:
+        patterns:
+         - "*"
+    versioning-strategy: 'increase'
```
Comment (senekor) [inline .github/dependabot.yml:14]: `/local-registry` is likely going to be detected as a library by dependabot, in which case it doesn't update as aggressively as for applications. This forces the most eager update strategy.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+updates:
+  - package-ecosystem: 'cargo'
+    directory: '/local-registry'
+    schedule:
+      interval: 'weekly'
+    groups:
+      - all:
+        patterns:
+         - "*"
```
Comment (senekor) [inline .github/dependabot.yml:13]: This should reduce the noise, maximum 1 PR per week. I see no reason to update separate crate versions in individual PRs.

**Comment by senekor** [comment]
@dhovart thanks for doing most of the work here a while back, feel free to chime in :slightly_smiling_face:

**Comment by dhovart** [comment]
> @dhovart thanks for doing most of the work here a while back, feel free to chime in 🙂

Thanks for taking care of this!  And continuing where I left.
I'm not familiar with configuring dependabot but with your explanations looks good to me :)

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**